src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,145 @@
+            followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+
+    }
+
+    public Command getSetTargetArmAngleCommand(Rotation2d angle) {
+        return new FunctionalCommand(
+                () -> generateAngleMotorProfile(angle),
+                this::setTargetAngleFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    public Command getSetTargetElevatorPositionCommand(double targetElevatorPosition) {
+        return new FunctionalCommand(
+                () -> generateElevatorMotorProfile(targetElevatorPosition),
+                this::setTargetElevatorPositionFromProfile,
+                (interrupted) -> {
+                },
+                () -> false,
+                this
+        );
+    }
+
+    private void generateAngleMotorProfile(Rotation2d targetAngle) {
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAngleEncoderPosition().getDegrees(), getAngleMotorVelocityDegreesPerSecond())
+        );
+        lastAngleMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void generateElevatorMotorProfile(double targetElevatorPosition) {
+        elevatorMotorProfile = new TrapezoidProfile(
+                ArmConstants.ELEVATOR_CONSTRAINTS,
+                new TrapezoidProfile.State(targetElevatorPosition, 0),
+                new TrapezoidProfile.State(getElevatorMotorPositionRevolutions(), getElevatorMotorVelocityRevolutionsPerSecond())
+        );
+        lastElevatorMotorProfileGenerationTime = Timer.getFPGATimestamp();
+    }
+
+    private void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            stopAngleMotors();
+            return;
+        }
+        ArmConstants.ANGLE_ENCODER_POSITION_SIGNAL.setUpdateFrequency(100);
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleMotorProfileTime());
+        double pidOutput = ArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAngleEncoderPosition().getDegrees(),
+                targetState.position
+        );
+        double feedForward = ArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                Units.degreesToRadians(targetState.velocity)
+        );
```

Comment:
Extract this into its own function called something like `calculateAngleMotorOutput`